engine/runtime/src/main/java/org/enso/interpreter/runtime/error/Warning.java

```diff
@@ -54,7 +54,10 @@ boolean hasSourceLocation() {
     }
 
     @ExportMessage
-    SourceSection getSourceLocation() {
+    SourceSection getSourceLocation() throws UnsupportedMessageException {
+      if (location == null) {
+        throw UnsupportedMessageException.create();
```

Comment:
First of all `Context.get(null).getBuiltins().nothing()` has wrong type - the return type is `SourceSection` and that return type is required by the InteropLibrary.

But more importantly, this is required by the InteropLibrary contract (and checked there):
```
@throws UnsupportedMessageException if and only if {@link #hasSourceLocation(Object)} returns
     *             <code>false</code> for the same receiver.
```
the _if and only if_ is checked using assertions - if `hasSourceLocation` returns false, the assertions check that `getSourceLocation` will actually throw this exception.

As for tests - again, running the project with assertions enabled will trigger this, because just any call to `hasSourceLocation` with assertions enabled performs that check. So I'm not sure if it's worth duplicating that logic, if just any call is basically a test for that. And we do have such a call in our unit tests.